File: Core/src/main/java/dev/lynith/core/ui/theme/AbstractTheme.java
package dev.lynith.core.ui.theme;

import dev.lynith.core.ui.components.impl.Button;
import dev.lynith.core.ui.theme.base.ButtonStyles;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.Setter;

@Getter @Setter @AllArgsConstructor
public abstract class AbstractTheme {

    private final String name;
    private final ThemeType type;

    public abstract ButtonStyles getButtonStyles(Button button);

    public enum ThemeType {
        DARK,
        LIGHT,
        HIGH_CONTRAST;
    }

}

File: Core/src/main/kotlin/dev/lynith/core/ui/theme/AbstractTheme.kt
package dev.lynith.core.ui.theme

import dev.lynith.core.ui.components.Component
import dev.lynith.core.ui.styles.ComponentStyles

abstract class AbstractTheme constructor(
    var name: String,
    var type: ThemeType
) {

    abstract val colorScheme: AbstractColorScheme

    abstract var componentMap: Map<Class<out Component<*, *>>, ComponentStyles<*, *>>

    enum class ThemeType {
        DARK,
        LIGHT,
        HIGH_CONTRAST
    }

}
